watchtower/__init__.py

```diff
@@ -91,35 +94,36 @@ def _submit_batch(self, batch, stream_name):
             # TODO: make this configurable/non-fatal
             raise Exception("Failed to deliver logs: {}".format(response))
 
-        self.sequence_tokens[stream_name] = response["nextSequenceToken"]
+        self.sequence_tokens[self.stream_name] = response["nextSequenceToken"]
 
     def emit(self, message):
-        stream_name = message.name
-        if stream_name not in self.sequence_tokens:
+        if self.stream_name is None:
+            self.stream_name = message.name
```

Comment:
This is still not correct. If you use the log handler with the stream name set to None, then this will permanently set the stream name to the name of the first logger whose message is passed, and incorrectly route all messages to that stream.

The solution is to avoid setting `self.stream_name` in this method: `stream_name = self.stream_name or message.name` will suffice, and then use stream_name through the rest of the method.

Because this class is expected to handle multiple stream names simultaneously in normal operation, it's also necessary to keep passing the stream name to batch_sender and _submit_batch, instead of having them get it through the class.